frontend/src/metabase/components/InputWithSelectPrefix/InputWithSelectPrefix.tsx

```diff
@@ -0,0 +1,127 @@
+import type { ChangeEvent } from "react";
+import { Component } from "react";
+import type { SelectChangeEvent } from "metabase/core/components/Select";
+import Select, { Option } from "metabase/core/components/Select";
+import { SelectPrefixInput } from "./InputWithSelectPrefix.styled";
+
+function splitValue({
+  value,
+  prefixes,
+  defaultPrefix,
+  caseInsensitivePrefix = false,
+}: {
+  value?: string;
+  prefixes: string[];
+  defaultPrefix: string;
+  caseInsensitivePrefix?: boolean;
+}): string[] {
+  if (value == null) {
+    return ["", ""];
+  }
+
+  const prefix = prefixes.find(
+    caseInsensitivePrefix
+      ? p => value.toLowerCase().startsWith(p.toLowerCase())
+      : p => value.startsWith(p),
+  );
+
+  return prefix ? [prefix, value.slice(prefix.length)] : [defaultPrefix, value];
+}
+
+type InputWithSelectPrefixProps = {
+  value?: string;
+  prefixes: string[];
+  defaultPrefix: string;
+  caseInsensitivePrefix?: boolean;
+  onChange: (e: { target: { value: string } }) => void;
+  placeholder?: string;
+  autoFocus?: boolean;
+  id?: string;
+};
+
+// eslint-disable-next-line import/no-default-export
+export default class InputWithSelectPrefix extends Component<
+  InputWithSelectPrefixProps,
+  { prefix: string; rest: string }
+> {
+  constructor(props: InputWithSelectPrefixProps) {
+    super(props);
+
+    const [prefix, rest] = splitValue(props);
+    this.state = { prefix, rest };
+  }
+
+  setPrefixAndRestFromValue() {
+    const { value } = this.props;
+
+    if (value) {
+      const [prefix, rest] = splitValue(this.props);
+      this.setState({ prefix, rest });
+    }
+  }
+
+  componentDidUpdate(prevProps: InputWithSelectPrefixProps) {
+    if (prevProps.value !== this.props.value) {
```

Comment:
Previously the onChange was called here, checking if the state changed.
I removed it as it's pretty much the same [as this anti-pattern](https://react.dev/learn/you-might-not-need-an-effect#sharing-logic-between-event-handlers).
(It also gave me some practical issue , i don't remember if it was with the not calling onChange with an empty `rest` or with the manual prefix handling)